demystifying_regular_expressions/demystifying_regular_expressions.md

```diff
@@ -0,0 +1,331 @@
+
+* `http` : The literal characters "http"
+* `s?`: Optionally, the letter "s"
+* `:\/\/` : A colon and two forward slashes: "://"
+* `([A-Za-z0-9\-]+\.)?`: Optionally, a group of characters consisting of: 
+
+  - `[A-Za-z0-9\-]+` : One or more characters that's either in the A-Z range, the a-z range, the 0-9 range, or a literal hyphen (`-`)
+  - `\.` : A literal period (`.`)
+* `[A-Za-z0-9\-]+`: One or more characters that's either in the A-Z range, the a-z range, the 0-9 range, or a literal hyphen (`-`)
+* `\.` : A literal period (`.`)
+* `[A-Za-z0-9\-\.]+`: One or more characters that's either in the A-Z range, the a-z range, the 0-9 range, a literal hyphen (`-`), or a literal period (`.`)
+* `\/?`: Optionally, a final forward slash
+
+
+## A Great Tool to Start With
+
+As a final step in demystifying regular expressions, it can be helpful to introduce the online regex checker as a great resource to learn and to check your regex before trying to learn how to make your own regular expressions or use regex in SQL, R, or Python code you're writing.
+
+We like "Regex 101", at https://www.regex101.com, but you can also search online for "regex checker" to find other, similar websites.  Regex 101 allows you to enter your regular expression as well as some sample text that you want to apply the regular expression to.  
+
+Below, or by [trying it out yourself](https://regex101.com/r/QUok4D), you can see that we put in our "valid URL" regular expression (at the top of the screen, in the appropriately labeled box) as well as the different varieties of URLs, some of which we think are valid and match our pattern, and some of which we think are invalid and should fail to match our pattern.
+
+If our test strings match the pattern, they are highlighted, and if they do not match, they are not highlighted.
+
+![Regex 101 application with sample URL regex pattern and test URL strings.](media/url_regex.png)<!-- style = "border: 1px solid rgb(var(--color-highlight));" -->
+
+This tool is useful, because sometimes you may not need to construct a regular expression, because it was given to you in a code sample you found online, or in code someone else wrote, like a coworker in your lab.  You will still want to test that it works the way you expect, however, even if you don't totally understand what each character of the regular expression means.  
+
+## Quiz: Demystifying Regex
+
+Please check all of the true statements below:
+
+[[ ]] Regular expressions are a way to detect statistical outliers in data.
+[[X]] Regular expressions are a way to define rules that describe a pattern in text.
+[[X]] Regular expressions can be used in multiple coding languages.
+[[ ]] The only way to check the accuracy of a regular expression someone gives you is by writing code. 
+[[X]] Regular expressions can be used to extract useful substrings.
+
+
+***
+<div class = "answer">
+
```

Comment:
This div box is rendering incorrectly... I think removing the white space between the last option and the `***` might fix it?